lib/krikri/harvesters/oai_harvester.rb

```diff
@@ -207,5 +218,31 @@ def record_xml(rec)
         .add_namespace_definition(nil, 'http://www.openarchives.org/OAI/2.0/')
       doc.to_xml
     end
+
+    private
+
+    def get_identifier(record)
+      return record.header.identifier unless @id_path
+
+      xml = Nokogiri.XML(record_xml(record))
```

Comment:
Yeah, I thought about that a bit, but didn't come up with anything that seemed much better.

I guess we could add an `as_doc: false` option to `#record_xml` so it can return the altered Nokogiri document.  Does that sound like a good fix?